src/panic.md

```diff
@@ -0,0 +1,95 @@
+# Panic
+
+Rust provides the ability to "panic" upon encountering an error that is not
+part of a function's signature; such an error is typically not expected to be
+recoverable within the context in which the error is encountered.
+
+Some language constructs, such as out-of-bounds [array indexing], panic
+automatically. There are also language features that provide a level of control
+over panic behavior:
+* A [_panic runtime_](#panic-runtimes) defined how a panic is handled during
+  runtime.
+* [FFI ABIs](items/functions.md#unwinding) may alter how panics behave.
+
+> **Note**: The Standard Library provides the capability to explicitly panic
+> via the [`panic!` macro][macro-panic].
+
+## Unwinding
+
+Panicking may either be recoverable or non-recoverable, though its behavior
+must be uniform throughout program execution. A recoverable panic "unwinds"
```

Comment:
As you noted in the catch_unwind PR, this is indeed not correct. Some panics, i.e. the one caused by "unwinding out of a no-unwind function", "unwinding out of a drop called during cleanup", as well as some UB-catching debug assertions in the standard library, are not recoverable even in `-Cpanic=unwind` builds.

Reply:
I've clarified that `panic=abort` forces all panics to be non-recoverable, whereas `panic=unwind` only guarantees that standard-library panic mechanisms, including `panic!` itself, are recoverable. Is that accurate?